client/blocks/comment-detail/index.jsx

```diff
@@ -82,16 +82,31 @@ export class CommentDetail extends Component {
 
 	toggleApprove = () => {
 		const { commentId, commentStatus, postId, setCommentStatus } = this.props;
-		setCommentStatus( commentId, postId, 'approved' === commentStatus ? 'unapproved' : 'approved' );
+		const isPersisted = 'approved' === commentStatus || 'unapproved' === commentStatus;
```

Comment:
maybe `shouldPersist` could more closely capture our intention here. it hasn't _yet_ persisted, after all, has it?